crates/router/src/core/payouts.rs

```diff
@@ -953,6 +953,22 @@ pub async fn call_connector_payout(
     let payout_attempt = &payout_data.payout_attempt.to_owned();
     let payouts = &payout_data.payouts.to_owned();
 
+    // fetch mca_id if not present
+    if payout_data.payout_attempt.merchant_connector_id.is_none() {
+        let merchant_connector_account = construct_profile_id_and_get_mca(
```

Comment:
This would be redundant db fetch, as we fetch the mca again during contruction of router_data

Reply:
Can we avoid fetching the mca in router data construction, and send the fetched mca from here?

Reply:
Made the changes